src/pybind/mgr/cli_api/module.py

```diff
@@ -1,92 +1,121 @@
-                osd_map, osd_map_tree, osd_map_crush, config, mon_map, fs_map,
-                osd_metadata, pg_summary, io_rate, pg_dump, df, osd_stats,
-                health, mon_status, devices, device <devid>, pg_stats,
-                pool_stats, pg_ready, osd_ping_times.
-        Note:
-            All these structures have their own JSON representations: experiment
-            or look at the C++ ``dump()`` methods to learn about them.
+        Generate CephCommand signature (dict-like)
         '''
-        t1_start = time.time()
-        str_arg = self.get(arg)
-        t1_end = time.time()
-        time_final = (t1_end - t1_start)
-        return HandleCommandResult(0, json.dumps(str_arg), str(time_final))
-
-    @CLICommand('mgr api benchmark get')
-    def api_get_benchmark(self, arg: str, number_of_total_calls: int,
-                          number_of_parallel_calls: int):
-        benchmark_runner = ThreadedBenchmarkRunner(number_of_total_calls, number_of_parallel_calls)
-        benchmark_runner.start(partial(self.get, arg))
-        benchmark_runner.join()
-        stats = benchmark_runner.get_stats()
-        return HandleCommandResult(0, json.dumps(stats), "")
-
-
-class ThreadedBenchmarkRunner:
-    def __init__(self, number_of_total_calls, number_of_parallel_calls):
-        self._number_of_parallel_calls = number_of_parallel_calls
-        self._number_of_total_calls = number_of_total_calls
-        self._threads = []
-        self._jobs: Queue = Queue()
-        self._time = 0.0
-        self._self_time = []
-        self._lock = threading.Lock()
-
-    def start(self, func):
-        if(self._number_of_total_calls and self._number_of_parallel_calls):
-            for thread_id in range(self._number_of_parallel_calls):
-                new_thread = threading.Thread(target=ThreadedBenchmarkRunner.timer,
-                                              args=(self, self._jobs, func,))
-                self._threads.append(new_thread)
-            for job_id in range(self._number_of_total_calls):
-                self._jobs.put(job_id)
-            for thread in self._threads:
-                thread.start()
+        return {
+            'prefix': f'mgr cli {self.func.__name__}',
+            'perm': API.perm.get(self.func)
+        }
+
+
+class MgrAPIReflector(type):
+    '''
+    Metaclass to register COMMANDS and Command Handlers via CLICommand
+    decorator
+    '''
+
+    def __new__(cls, name, bases, dct):
+        klass = super().__new__(cls, name, bases, dct)
+        cls.threaded_benchmark_runner = None
+        for base in bases:
+            for name, func in inspect.getmembers(base, cls.is_public):
+                # However not necessary (CLICommand uses a registry)
+                # save functions to klass._cli_{n}() methods. This
+                # can help on unit testing
+                wrapper = cls.func_wrapper(func)
+                if API.cliapi.get(func):
```

Comment:
We shouldn't need a new decorator here, the `is_public()` method should filter out the methods you don't want to expose to the CLI (e.g.: API hooks, internal methods and private methods). If there's any method left that we don't want exposed, we should follow the approach of excluding methods rather than cherry-picking the ones we want in.